independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/ResteasyReactiveOutputStream.java

```diff
@@ -233,12 +236,24 @@ private void prepareWrite(ByteBuf buffer, boolean finished) throws IOException {
                 } else {
                     context.serverResponse().setResponseHeader(HttpHeaderNames.CONTENT_LENGTH, "" + buffer.readableBytes());
                 }
-            } else if (!request.response().headers().contains(HttpHeaderNames.CONTENT_LENGTH)) {
+            } else if (!contentLengthSet()) {
                 request.response().setChunked(true);
             }
         }
     }
 
+    private boolean contentLengthSet() {
+        if (request.response().headers().contains(HttpHeaderNames.CONTENT_LENGTH)) {
+            return true;
+        }
+        LazyResponse lazyResponse = context.getResponse();
+        if (!lazyResponse.isCreated()) {
+            return false;
+        }
+        MultivaluedMap<String, Object> responseHeaders = lazyResponse.get().getHeaders();
+        return (responseHeaders != null) && responseHeaders.containsKey(HttpHeaders.CONTENT_LENGTH);
```

Comment:
You're using `HttpHeaderNames` for one and `HttpHeaders` for the other. Is it normal (just checking)?

Reply:
Yes, it's expected. The former is from `Netty` and is of type `CharSequence` (which Vert.x expects) while the latter comes from JAX-RS and is of type `String` (which is what JAX-RS use everywhere)/